backend/lib/services/terminals/index.js

```diff
@@ -221,17 +225,21 @@ async function getTargetCluster ({ user, namespace, name, target, preferredHost,
       } else {
         const projectName = findProjectByNamespace(namespace).metadata.name
         targetCluster.namespace = namespace
-        const serviceAccountName = 'dashboard-webterminal'
+        const serviceAccountName = DASHBOARD_WEBTERMINAL_NAME
 
-        // test if service account exists
-        await client.core.serviceaccounts.get(namespace, serviceAccountName)
+        // test if service account exists and is not in deletion
+        const serviceAccount = await client.core.serviceaccounts.get(namespace, serviceAccountName)
+        if (serviceAccount.metadata?.deletionTimestamp) {
```

Comment:
If a serviceAccount is returned the metadata cannot be nullish, I think?

Reply:
✅